.github/workflows/build-and-push-image.yml

```diff
@@ -91,3 +91,25 @@ jobs:
             } catch (error) {
               core.setFailed(error.message);
             }
+
+      - name: Trigger Build and Push NuGet Package workflow
```

Comment:
```
if: needs.set-env.outputs.environment == 'production'
```

should be doable here, that way your script is only doing one thing which is triggering the next workflow :)

```diff
- const environment = "${{ needs.set-env.outputs.environment }}";
- if (environment !== "production") {
-    console.log("Not a production deployment. Skipping trigger.");
-    return;
- }
```

Reply:
Ah nice, will update